### PR TITLE
Refactor and extend execution environment functionalities

### PR DESCRIPTION
**Pull request by T-Haller**
#### Short Description
Refactoring and extension of software platform, cloud service and compute resource functionalities

#### Proposed Changes

  * Renamed all Backend related classes and functions to ComputeResource
  * Renamed ComputingResourceProperty to ComputeResourceProperty
  * Reinserted CloudServices and ComputeResources
  * Reworked controllers of software platform, cloud service and compute resource
  * Reworked services of software platform, cloud service and compute resource

Implements: PlanQK/q-tal#67 PlanQK/q-tal#62

### CODE REVIEW

File: docs/dev/OpenAPI.md [hunk 1]
```diff
@@ -9,4 +9,4 @@ An example of the generated code can be found in the [qc-atlas-ui project](https
 ## How do I generate API client libraries?
 Pick a generator for the programming language / framework of your choice, for example from [here](https://github.com/OpenAPITools/openapi-generator).
 The generator used in the [qc-atlas-ui project](https://github.com/PlanQK/qc-atlas-ui/tree/master/generated/api) can be found in this [repository](https://github.com/cyclosproject/ng-openapi-gen). 
-To obtain the OpenAPI file launch the backend, the file is located in `/atlas/v3/api-docs`.
\ No newline at end of file
+To obtain the OpenAPI file launch the computeResource, the file is located in `/atlas/v3/api-docs`.
```
Comment (FlxB2) [inline docs/dev/OpenAPI.md:12]: I think something went wrong here after batch refactoring :smile:

Comment (T-Haller) [reply docs/dev/OpenAPI.md]: Should be fixed now :)

**Comment by c-mueller** [comment]
Retry DeepCode

File: org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/utils/ListParametersMethodArgumentResolver.java [hunk 2]
```diff
@@ -43,6 +45,10 @@ public Object resolveArgument(MethodParameter methodParameter, @Nullable ModelAn
         Sort sort = sortResolver.resolveArgument(methodParameter, mavContainer, webRequest, binderFactory);
         Pageable pageable = getPageable(methodParameter, page, pageSize);
 
+        if ((page != null && pageSize != null) && (page.equals("-1") && pageSize.equals("-1"))) {
```
Comment (timniederhausen) [inline org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/utils/ListParametersMethodArgumentResolver.java:48]: the question is: do we have any use-cases for unpaged collections? and shouldn't this behavior be scoped to those exceptions? (e.g. via the @PageableDefault annotation)

in any case, we can do either of those:

* call `setFallbackPageable(Pageable.unpaged())` in the constructor, meaning any request without page / size returns an unpaged Pageable.
* keep this
* remove it and set explicit defaults where we need unpaged collections (imho the best option)

**Pull request merged**